### PR TITLE
dir restructure to support more CGP

### LINKED ISSUE #1240: `polkadot-parachains` folder restructuring.
In order to enhance the developer experience and documentation for engineers working with the parachains living under  [polkadot-parachains](https://github.com/paritytech/cumulus/tree/master/polkadot-parachains), a new folder called `parachain-runtimes` will be created with the following structure:

- parachain-runtimes
    - chain-specs (formerly known as `res`) 
    - common
        - utilities (common shared code / libraries)
        - <del>tests ( folder containing unit tests for different functionalities )</del> Moved to`polkadot-parachain` folder
    - pallets
        - specific pallets for parachains (e.g. collator-selection)
    - <del>src (entry point folder for commands/services etc )</del> Moved to `polkadot-parachain` folder
    - starters
        - shell (can be used to launch an "empty" runtime that can be upgraded via a downward message)
        - seedling (runtime with sudo to register prior to migrating a solo-chain into a parachain)
        - <del>scaffold (formerly known as `parachain-template`, a scaffold/template for teams to start from when developing a runtime)</del>  [Discussion](https://github.com/paritytech/cumulus/issues/1240#issuecomment-1120853289)
    - testing
        - pen-pal (parachain runtime ready with the necessary XCM setup to test interactions with Statemine/Statemint)
        - rococo-parachain (used for creating Tick/Trick/Track parachains). Discussion #1222
    - assets
        - statemine
        - statemint
     - contracts
        - substrate-contracts
    - governance (as an example of a potential common good parachain dedicated to governance)
        - kusama-gov
        - polkadot-gov
    - etc...

Also based as a result of a diagnostic of the existing parachains on the folder, Wesmint would be removed since we would be deploying the Statemint runtime directly on Westend.

Every parachain runtime should include a rust doc with the purpose of it.

@joepetrowski

### PR DESCRIPTION
**Pull request by gilescope**
Readying the ground for more CGP. Fixes #1240

### CODE REVIEW

File: polkadot-parachain/Cargo.toml [hunk 1]
```diff
@@ -27,7 +27,7 @@ statemint-runtime = { path = "../parachain-runtimes/assets/statemint" }
 statemine-runtime = { path = "../parachain-runtimes/assets/statemine" }
 westmint-runtime = { path = "../parachain-runtimes/assets/westmint" }
 canvas-kusama-runtime = { path = "../parachain-runtimes/smart-contracts/canvas-kusama" }
-jsonrpsee = { version = "0.13.0", features = ["server"] }
+jsonrpsee = { version = "0.13.1", features = ["server"] }
```
Comment (chevdor) [inline polkadot-parachain/Cargo.toml:30]:
```suggestion
jsonrpsee = { version = "0.13", features = ["server"] }
```
Would that not be simpler ? Do we really want the `0.13.1` and not the `0.13.2` (for instance) ?

**Review by chevdor — CHANGES_REQUESTED** [review]

File: polkadot-parachain/Cargo.toml (hunk 1 above)
Comment (niklasad1) [reply polkadot-parachain/Cargo.toml:30]: this is doesn't really matter `jsonrpsee 0.13.1` is just a doc fix on doc.rs :)

Comment (chevdor) [reply polkadot-parachain/Cargo.toml:30]: My comment is also about future versions. Any `0.13.x` should be compatible so we should take the latest available inseatd of pinning to `0.13.1`. `jsonrpsee` follows semver @niklasad1 right ?

Comment (niklasad1) [reply polkadot-parachain/Cargo.toml:30]: yes

File: docker/docker-compose.yml [hunk 2]
```diff
@@ -9,7 +9,7 @@ services:
     volumes:
       - "polkadot-data-alice:/data"
       - type: bind
-        source: ./test/parachain/res/polkadot_chainspec.json
+        source: ./test/parachain/chain-specs/polkadot_chainspec.json
         target: /chainspec.json
```
Comment (NukeManDan) [inline docker/docker-compose.yml:13]: I think this is a note sensible convention for this repo and Polkadot. If you agree, might we have another PR to change the name and update ci as needed for it?

Comment (gilescope) [reply docker/docker-compose.yml:13]: `res` is just plain confusing to the newcomer. If we have other resources we probably would not want them mingled up with the chain-spec files.

**Review by joepetrowski — APPROVED** [review]
Besides some minor linking errors and the `smart-contracts` vs. `contracts` question it LGTM.

File: Cargo.toml [hunk 3]
```diff
@@ -24,16 +24,17 @@ members = [
...
+	"parachain-runtimes/pallets/parachain-info",
+	"parachain-runtimes/pallets/ping",
+	"parachain-runtimes/testing/rococo-parachain",
+	"parachain-runtimes/starters/shell",
+	"parachain-runtimes/starters/seedling",
+	"parachain-runtimes/common",
+	"parachain-runtimes/assets/statemint",
+	"parachain-runtimes/assets/statemine",
+	"parachain-runtimes/assets/westmint",
+	"parachain-runtimes/smart-contracts/contracts-rococo",
```
[the remark this answers is not in the document]
Comment (cmichi) [reply Cargo.toml:37]: Yup, `contracts` please.

Context is that the pallet is also named `pallet-contracts` and the development node we have `substrate-contracts-node`. So we want to keep it consistent.

File: docker/docker-compose.yml (hunk 2 above)
Comment (gilescope) [reply docker/docker-compose.yml:13]: Here you go Dan: https://github.com/paritytech/polkadot/pull/5570

**Review by hbulgarini — APPROVED** [review]
After several back and forth, the changes looks good to me.

**Review by chevdor — APPROVED** [review]
LGTM

**Review by sergejparity — APPROVED** [review]

**Review by rcny — APPROVED** [review]

**Comment by gilescope** [comment]
bot merge

**Comment by gilescope** [comment]
bot merge

**Pull request merged**